### PR TITLE
Integrate mibc mono config extension

### PR DESCRIPTION
**Pull request by mdh1418**
Follow up to https://github.com/dotnet/runtime/pull/71359.

With .mibc profiles now containing a MibcConfig, there are a couple of places that can use some modifications to better integrate this change.
1) ClrEtwAll.man runtime sku - Only knows DesktopClr and CoreClr. Needs to be extended to know Mono (shows up as `4`)
2) Mono AOT Compiler - Can leverage the MibcConfig to skip .mibc's generated from non-mono runtimes.

### This PR makes the following changes:

- Extends mono aot compiler to check the compatibility of the provided .mibc profile based on its MibcConfig
- Extends ClrEtwAll.man to recognize `0x4` as `"Mono"` (follow up on https://github.com/dotnet/runtime/pull/71359#issuecomment-1173445310)
- Follows up on https://github.com/dotnet/runtime/pull/70194#discussion_r891397446

### CODE REVIEW

**Comment by mdh1418** [comment]
It looks like the Runtime value still shows up as `4` instead of the desired `"Mono"`. Are there more areas I should extend the runtime Sku in dotnet/runtime @brianrob ? Where else needs the Sku extension for Perfview, Visual Studio, etc to recognize the new sku value of `0x4`?

**Comment by brianrob** [comment]
@mdh1418, your change to add 0x4 to the `RuntimeSkuMap` looks correct, though I don't see anything that emits this value in the current change.  Is that elsewhere?

With regard to making the value "Mono" show up properly, you will want to update https://github.com/microsoft/perfview/blob/main/src/TraceEvent/Parsers/ClrTraceEventParser.cs#L12694.  This will cover PerfView and TraceEvent, and then any tools that pick-up new versions of TraceEvent.

**Comment by mdh1418** [comment]
@brianrob 
> emits this value in the current change. Is that elsewhere?

Yes, I believe the Sku value 0x4 makes it to the .nettrace through a rundown event `FireEtwRuntimeInformationDCStart` https://github.com/dotnet/runtime/blob/cc0ccbe1dd1c7f890ab3e42815fc87d5b6ce94f7/src/mono/mono/eventpipe/ep-rt-mono.c#L2872 and from there, it makes it to the .mibc through https://github.com/dotnet/runtime/blob/cc0ccbe1dd1c7f890ab3e42815fc87d5b6ce94f7/src/coreclr/tools/dotnet-pgo/Program.cs#L1776-L1777. These changes were in a previous PR #71359 

Thanks, I'll make changes to ClrTraceEventParser and see how it reflects locally!

**Comment by brianrob** [comment]
@mdh1418 sounds good.  What you've described sounds right to me.  You will want to make the change to TraceEvent locally and then patch in the updated DLL where dotnet-pgo picks it up.  You can do this all locally, and assuming it works, then you'd just need to take a new version of TraceEvent into dotnet-pgo.

**Comment by mdh1418** [comment]
@brianrob I was able to verify that the MibcConfig section of the .mibc profile reflected `Mono` under Runtime after I built perfview locally with the RuntimeSku extension there.
Screen Shot 2022-07-05 at 5 57 04 PM

I can put up a PR on perfview with the single line change and we can update the TraceEventVersion used in this repo to whichever version that PR makes it to.

**Comment by brianrob** [comment]
Sounds great!

**Review by lateralusX — APPROVED** [review]
LGTM! Couple of nits.

File: src/mono/mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -13350,14 +13350,116 @@ add_mibc_group_method_methods (MonoAotCompile *acfg, MonoMethod *mibcGroupMethod
...
+	// If there is no MibcConfig, assume it was a .mibc generated prior to MibcConfig addition
+	if (!mibcConfig)
+		return TRUE;
+
+	MonoMethodHeader *mibcConfigHeader = mono_method_get_header_internal (mibcConfig, error);
+	mono_error_assert_ok (error);
+
+	gboolean isConfigCompatible = FALSE;
+	MibcConfigParserState state = PARSING_MIBC_CONFIG_NONE;
+	uint8_t *cur = (uint8_t*)mibcConfigHeader->code;
```
[the remark this answers is not in the document]
Comment (mdh1418) [reply src/mono/mono/mini/aot-compiler.c:13418]: Thanks! When first implementing the IL iterator loop I was fixed on using `uint8_t` from an initial suggestion and focused on making that work. I modified the other two IL iterator loops to conform to this suggestion as well

**Pull request merged**